model/src/main/scala/za/co/absa/atum/model/dto/package.scala

```diff
@@ -28,4 +27,7 @@ package object dto {
   implicit val decodeAdditionalDataDTO: Decoder[AdditionalDataDTO] = Decoder.decodeMap[String, Option[String]]
   implicit val encodeAdditionalDataDTO: Encoder[AdditionalDataDTO] = Encoder.encodeMap[String, Option[String]]
 
+  // Implicit encoders and decoders for PartitioningDTO
```

Comment:
I think it's not the best practice to define such implicits in package object as by doing that you implicitly import them into every possible scope in the entire package which might leaad to unintended implicit conversions or resolutions. These types should be changed. AdditionalDataDTO will undergo some changes when working on endpoints related to AdditionalData. PartitioningDTO could be modeled as case class and json serde defined in it's companion.